cmd/handler-utils.go

```diff
@@ -424,6 +424,14 @@ func extractAPIVersion(r *http.Request) string {
 	return regexVersion.FindString(r.URL.Path)
 }
 
+func methodNotAllowedHandler(w http.ResponseWriter, r *http.Request) {
+	writeErrorResponseString(r.Context(), w, APIError{
+		Code:           "XMinioPeerVersionMismatch",
```

Comment:
This code is used for all handlers and errors, you should perhaps add context here per API? such as `XMinioAdminVersionMismatch`,..

`XMinioS3APIMismatch` ?